component/rules.jsonnet

```diff
@@ -252,44 +252,149 @@ local additionalRules = {
-  annotateRules +
-  filterRules +
-  {
-    _config+:: {
-      namespace: params.namespace,
+local common_mixins = (import 'kubernetes-mixin/alerts/add-runbook-links.libsonnet') +
+                      alterRules +
+                      annotateRules +
+                      filterRules;
 
-      prometheus+:: {
-        name: params.prometheusInstance,
+local common_config = {
+  prometheusRule+: {
+    metadata+: {
+      name: params.prometheusInstance + '-' + super.name,
+    },
+  },
+  _config+:: {
+    namespace: params.namespace,
+    mixin+: {
+      ruleLabels+: {
+        prometheus: params.prometheusInstance,
       },
+    },
+  },
+  mixin+:: common_mixins,
+};
 
-      alertmanager+:: {
-        name: params.alertmanagerInstance,
-      },
 
-      namespaceSelector: params.alerts.namespaceSelector,
-    },
-  };
+local kube_prometheus_number = std.parseJson(std.strReplace(params.jsonnetfile_parameters.kube_prometheus_version, 'release-', ''));
 
-local kp =
-  commonkp +
-  if params.rancher_monitoring_version == 'v1' then
-    {
-      _config+:: {
-        prometheusOperatorSelector: 'job="expose-operator-metrics",namespace="cattle-prometheus"',
-        kubeApiserverSelector: 'job="kubernetes"',
-        kubeStateMetricsSelector: 'job="expose-kubernetes-metrics"',
-        kubeletSelector: 'job="expose-kubelets-metrics"',
-        nodeExporterSelector: 'job="expose-node-metrics"',
-      },
-    }
-  else
-    {
-    };
+// for rancher-monitoring prometheus > 0.7
+local kp2 = (import 'kube-prometheus/main.libsonnet') +
+            (import 'kube-prometheus/addons/managed-cluster.libsonnet') +
+            {
+              kubeStateMetrics+:
+                common_config
+                {
+                  mixin+:: {
+                    _config+:: if params.rancher_monitoring_version == 'v1' then
+                      {
+                        kubeStateMetricsSelector: 'job="expose-kubernetes-metrics"',
+                      }
+                    else {},
+                  },
+                },
+              nodeExporter+:
+                common_config
+                {
+                  mixin+:: {
+                    _config+:: if params.rancher_monitoring_version == 'v1' then
+                      {
+                        nodeExporterSelector: 'job="expose-node-metrics"',
+                      }
+                    else {},
+                  },
+                },
+              prometheus+:
+                common_config
+                {
+                  mixin+:: additionalRules,
+                  prometheusRule+: {
+                    metadata+: {
+                      name: params.prometheusInstance + '-prometheus-rules',
+                    },
+                  },
+                },
+              prometheusOperator+:
+                common_config
+                {
+                  mixin+:: {
+                    _config+:: if params.rancher_monitoring_version == 'v1' then
```

Comment:
Not all clusters using 1.20 use Rancher Monitoring V2. Ignoring it would break them.
I think it is better to support it for now, and not add an additional constraint to the upgrades.